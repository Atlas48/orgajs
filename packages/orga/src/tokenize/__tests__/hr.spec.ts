import tok from './tok'

describe('tokenize hr', () => {
  it('knows horizontal rules', () => {
    expect(tok('-----')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----",
          "type": "hr",
        },
      ]
    `)
    expect(tok('------')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "------",
          "type": "hr",
        },
      ]
    `)
    expect(tok('--------')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "--------",
          "type": "hr",
        },
      ]
    `)
    expect(tok('  -----')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----",
          "type": "hr",
        },
      ]
    `)
    expect(tok('-----   ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----   ",
          "type": "hr",
        },
      ]
    `)
    expect(tok('  -----   ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----   ",
          "type": "hr",
        },
      ]
    `)
    expect(tok('  -----  \t ')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----  	 ",
          "type": "hr",
        },
      ]
    `)
  })

  it('knows these are not horizontal rules', () => {
    expect(tok('----')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "----",
          "type": "text.plain",
          "value": "----",
        },
      ]
    `)
    expect(tok('- ----')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-",
          "indent": 0,
          "ordered": false,
          "type": "list.item.bullet",
        },
        Object {
          "_text": "----",
          "type": "text.plain",
          "value": "----",
        },
      ]
    `)
    expect(tok('-----a')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----a",
          "type": "text.plain",
          "value": "-----a",
        },
      ]
    `)
    expect(tok('_-----')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "_-----",
          "type": "text.plain",
          "value": "_-----",
        },
      ]
    `)
    expect(tok('-----    a')).toMatchInlineSnapshot(`
      Array [
        Object {
          "_text": "-----    a",
          "type": "text.plain",
          "value": "-----    a",
        },
      ]
    `)
  })
})
