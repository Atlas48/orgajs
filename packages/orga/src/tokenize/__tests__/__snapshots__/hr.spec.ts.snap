// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize hr knows horizontal rules 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 5,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows horizontal rules 2`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows horizontal rules 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 8,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows horizontal rules 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 7,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows horizontal rules 5`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 8,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows horizontal rules 6`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows horizontal rules 7`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 11,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "hr",
  },
]
`;

exports[`tokenize hr knows these are not horizontal rules 1`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 4,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize hr knows these are not horizontal rules 2`] = `
Array [
  Object {
    "indent": 0,
    "ordered": false,
    "position": Object {
      "end": Object {
        "column": 1,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "list.item.bullet",
  },
  Object {
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 2,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize hr knows these are not horizontal rules 3`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize hr knows these are not horizontal rules 4`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 6,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;

exports[`tokenize hr knows these are not horizontal rules 5`] = `
Array [
  Object {
    "position": Object {
      "end": Object {
        "column": 10,
        "line": 0,
      },
      "start": Object {
        "column": 0,
        "line": 0,
      },
    },
    "type": "text.plain",
  },
]
`;
